packages/toolkit/src/view/knowledge/components/CreateKnowledgeBaseCard.tsx

```diff
@@ -1,153 +1,66 @@
-              onClick={onDelete}
-              className="!px-4 !py-2.5 !text-semantic-error-default product-body-text-4-medium"
-            >
-              <Icons.Trash01 className="mr-2 h-4 w-4 stroke-semantic-error-default" />
-              Delete
-            </DropdownMenu.Item>
-          </DropdownMenu.Content>
-        </DropdownMenu.Root>
-      </div>
-    </React.Fragment>
-  );
-};
 type CreateKnowledgeBaseCardProps = {
   knowledgeBase: KnowledgeBase;
   onCardClick: () => void;
-  onUpdateKnowledgeBase: (
-    data: EditKnowledgeDialogData,
-    kbId: string,
-  ) => Promise<void>;
+  onUpdateKnowledgeBase: (data: EditKnowledgeDialogData, kbId: string) => Promise<void>;
   onCloneKnowledgeBase: (knowledgeBase: KnowledgeBase) => Promise<void>;
-  onDeleteKnowledgeBase: (knowledgeBase: KnowledgeBase) => void;
+  onDeleteKnowledgeBase: (knowledgeBase: KnowledgeBase) => Promise<void>;
   disabled?: boolean;
 };
+
 const selector = (store: InstillStore) => ({
   accessToken: store.accessToken,
   enabledQuery: store.enabledQuery,
   selectedNamespace: store.navigationNamespaceAnchor,
 });
 
-export const CreateKnowledgeBaseCard = ({
+export const CreateKnowledgeBaseCard: React.FC<CreateKnowledgeBaseCardProps> = ({
   knowledgeBase,
   onCardClick,
   onUpdateKnowledgeBase,
   onCloneKnowledgeBase,
   onDeleteKnowledgeBase,
   disabled = false,
-}: CreateKnowledgeBaseCardProps) => {
+}) => {
   const [deleteDialogIsOpen, setDeleteDialogIsOpen] = React.useState(false);
   const [editDialogIsOpen, setEditDialogIsOpen] = React.useState(false);
-  const [isHovered, setIsHovered] = React.useState(false);
-  const [mousePosition, setMousePosition] = React.useState({ x: 0, y: 0 });
   const [isMenuOpen, setIsMenuOpen] = React.useState(false);
+  const cardRef = React.useRef<HTMLDivElement>(null);
 
-  const { accessToken, enabledQuery, selectedNamespace } = useInstillStore(
-    useShallow(selector),
-  );
+  const { accessToken, enabledQuery, selectedNamespace } = useInstillStore(useShallow(selector));
 
   const { data: filesData } = useListKnowledgeBaseFiles({
     namespaceId: selectedNamespace || null,
     knowledgeBaseId: knowledgeBase.catalogId,
     accessToken: accessToken || null,
-    enabled: enabledQuery && isHovered,
+    enabled: enabledQuery,
   });
 
   const chunkQueries = useQueries({
     queries: (filesData?.files || []).map((file) => ({
       queryKey: ["chunks", knowledgeBase.catalogId, file.fileUid],
-      queryFn: () =>
-        getAllChunks(
-          accessToken || "",
-          knowledgeBase.ownerName,
-          knowledgeBase.catalogId,
-          file.fileUid,
-        ),
-      enabled: enabledQuery && isHovered && !!filesData && !!accessToken,
+      queryFn: () => getAllChunks(
+        accessToken || "",
+        knowledgeBase.ownerName,
+        knowledgeBase.catalogId,
+        file.fileUid,
+      ),
+      enabled: enabledQuery && !!filesData && !!accessToken,
     })),
   });
```

Comment:
@thewbuk We don't allow in component react-query written like this, if you need this query function, you should put it in the react-query-service folder as separated file